src/chainstate/stacks/db/mod.rs

```diff
@@ -840,20 +846,49 @@ impl StacksChainState {
     }
 
     /// Begin processing an epoch's transactions within the context of a chainstate transaction
-    pub fn chainstate_block_begin<'a>(chainstate_tx: &'a ChainstateTx<'a>, clarity_instance: &'a mut ClarityInstance, parent_burn_hash: &BurnchainHeaderHash, parent_block: &BlockHeaderHash, new_burn_hash: &BurnchainHeaderHash, new_block: &BlockHeaderHash) -> ClarityTx<'a> {
+    pub fn chainstate_block_begin<'a>(chainstate_tx: &'a ChainstateTx<'a>,
+                                      clarity_instance: &'a mut ClarityInstance,
+                                      parent_burn_hash: &BurnchainHeaderHash,
+                                      parent_block: &BlockHeaderHash,
+                                      new_burn_hash: &BurnchainHeaderHash,
+                                      new_block: &BlockHeaderHash) -> ClarityTx<'a> {
+
         let conf = chainstate_tx.config.clone();
         StacksChainState::inner_clarity_tx_begin(conf, chainstate_tx.headers_tx.deref().deref(),
-                                                 clarity_instance, parent_burn_hash, parent_block, new_burn_hash, new_block)
+                                                 clarity_instance, parent_burn_hash, parent_block, new_burn_hash, new_block, None)
+    }
+    
+    /// Begin processing an epoch's transactions within the context of a chainstate transaction,
+    /// with a given computational budget
+    pub fn chainstate_block_begin_limited<'a>(chainstate_tx: &'a ChainstateTx<'a>,
```

Comment:
In https://github.com/blockstack/stacks-blockchain/pull/1412, which I'm about to merge (as soon as I figure out why the CircleCI tarpaulin build is failing), the block limit is set in the ClarityInstance, so it does not need to be (and no longer can be) passed in during a block begin -- rather it's set when the ClarityInstance opens. This avoids needing to pass the block limit around.

Reply:
Is there a way to set the block limit in a ClarityInstance?  The mining code needs to be able to set the block limit for an anchored block in order to support the miner's ability to decide how much of the budget will be split between the anchored block and microblock stream.

Reply:
We could make the block_limit field public, and then the ClarityInstance's block_limit could be adjusted.

Reply:
I see #1412 is in master already.  I can just make that change as part of this PR.

Reply:
Made the merge.